### PR TITLE
Improve list_parameters example

### PR DESCRIPTION
**Pull request by esteve**
This PR instantiates the parameters service before using the client and adds a few more cases.

Please review.

### CODE REVIEW

File: rclcpp_examples/src/parameters/list_parameters_async.cpp [hunk 1]
```diff
@@ -22,6 +22,9 @@ int main(int argc, char ** argv)
 
   auto node = rclcpp::Node::make_shared("list_parameters_async");
 
+  // TODO(esteve): Make the parameter service automatically start with the node.
+  auto parameter_service = std::make_shared<rclcpp::parameter_service::ParameterService>(node);
+
   auto parameters_client = std::make_shared<rclcpp::parameter_client::AsyncParametersClient>(node);
 
   // TODO(wjwwood): set more interesting parameters once the namespacing is figured out.
```
Comment (wjwwood) [inline rclcpp_examples/src/parameters/list_parameters_async.cpp]: Should this TODO be dropped now?

**Comment by wjwwood** [comment]
+1

File: rclcpp_examples/src/parameters/list_parameters_async.cpp (hunk 1 above)
Comment (esteve) [inline rclcpp_examples/src/parameters/list_parameters_async.cpp]: Good catch! Fixed.

**Comment by dirk-thomas** [comment]
+1

**Comment by tfoote** [comment]
+1

**Pull request merged**